lib/rules/lighter-video.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * @fileoverview Rule to avoid video 
+ * @author Aleksandra Borowska
+ */
+
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+/** @type {import('../shared/types').Rule} */
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Calculate carbon emissions and video duration for embedded YouTube videos',
+      recommended: true,
+      url: "https://ecolint.org/docs/rules/lighter-video"
+    },
+    schema: [
+
+    ]
+  },
+  create: function (context) {
+
+    const emissionsPerMinute = {
+      small: 0.8,
+      medium: 1.0,
+      large: 1.2,
+      hd720: 1.5,
+      hd1080: 2.0,
+      highres: 2.5,
+    };
+
+    function getPropertyValue(obj, property) {
+      const prop = obj.properties.find(prop => prop.key.name === property);
```

Comment:
for future reason i would add here optional chaining e.g.:
`obj?.properties?.find(prop => prop?.key?.name === name)`

Reply:
Ok, made changes with optional chaining.

Reply:
I see no changes here :/